vita3k/CMakeLists.txt

```diff
@@ -253,6 +253,7 @@ elseif(LINUX)
 		COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_BINARY_DIR}/external/discord_game_sdk/lib/x86_64/libdiscord_game_sdk.so" "$<TARGET_FILE_DIR:vita3k>")
 	endif()
 elseif(WIN32)
+	target_link_libraries(vita3k PRIVATE Dbghelp Version)
```

Comment:
Because if you leave it out it leads to errors like this
```
[build] FAILED: bin/Vita3K.exe 
[build] C:\WINDOWS\system32\cmd.exe /C "cd . && C:\msys64\clang64\bin\c++.exe -fsanitize=address -DTRACY_NO_CRASH_HANDLER -D_LIBCPP_ENABLE_ASSERTIONS=1 -g -fsanitize=address @CMakeFiles\vita3k.rsp -o bin\Vita3K.exe -Wl,--out-implib,lib\libVita3K.dll.a -Wl,--major-image-version,0,--minor-image-version,0 && C:\WINDOWS\system32\cmd.exe /C "cd /D E:\build-vita3k-clang-debug\vita3k && C:\msys64\clang64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/../data E:/build-vita3k-clang-debug/bin/data && C:\msys64\clang64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/../lang E:/build-vita3k-clang-debug/bin/lang && C:\msys64\clang64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/shaders-builtin E:/build-vita3k-clang-debug/bin/shaders-builtin && C:\msys64\clang64\bin\cmake.exe -E copy_if_different C:/src/Vita3K/external/sdl/windows/lib/x64/SDL2.dll E:/build-vita3k-clang-debug/bin && cd /D E:\build-vita3k-clang-debug\vita3k && C:\msys64\clang64\bin\cmake.exe -E copy_if_different C:/src/Vita3K/vita3k/script/update-windows.bat E:/build-vita3k-clang-debug/bin/update-vita3k.bat""
[build] ld.lld: error: undefined symbol: GetFileVersionInfoSizeExW
[build] >>> referenced by C:/src/Vita3K/vita3k/util/src/vc_runtime_checker.cpp:52
[build] >>>               vita3k/CMakeFiles/vita3k.dir/util/src/vc_runtime_checker.cpp.obj:(VCRuntimeCheckObject::VCRuntimeCheckObject())
[build] 
[build] ld.lld: error: undefined symbol: GetFileVersionInfoExW
[build] >>> referenced by C:/src/Vita3K/vita3k/util/src/vc_runtime_checker.cpp:61
[build] >>>               vita3k/CMakeFiles/vita3k.dir/util/src/vc_runtime_checker.cpp.obj:(VCRuntimeCheckObject::VCRuntimeCheckObject())
[build] 
[build] ld.lld: error: undefined symbol: VerQueryValueW
[build] >>> referenced by C:/src/Vita3K/vita3k/util/src/vc_runtime_checker.cpp:61
[build] >>>               vita3k/CMakeFiles/vita3k.dir/util/src/vc_runtime_checker.cpp.obj:(VCRuntimeCheckObject::VCRuntimeCheckObject())
[build] 
[build] ld.lld: error: undefined symbol: __declspec(dllimport) SymFromAddr
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:679
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DecodeCallstackPtr(unsigned long long))
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:500
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DecodeCallstackPtrFast(unsigned long long))
[build] 
[build] ld.lld: error: undefined symbol: __declspec(dllimport) SymGetLineFromAddr64
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:687
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DecodeCallstackPtr(unsigned long long))
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:603
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DecodeSymbolAddress(unsigned long long))
[build] 
[build] ld.lld: error: undefined symbol: __declspec(dllimport) SymInitialize
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:327
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DbgHelpInit())
[build] 
[build] ld.lld: error: undefined symbol: __declspec(dllimport) SymSetOptions
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:328
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DbgHelpInit())
[build] 
[build] ld.lld: error: undefined symbol: __declspec(dllimport) SymLoadModuleEx
[build] >>> referenced by C:/src/Vita3K/external/tracy/public/client/TracyCallstack.cpp:338
[build] >>>               libtracy.a(TracyClient.cpp.obj):(tracy::DbgHelpLoadSymbolsForModule(char const*, unsigned long long, unsigned int))
[build] c++: error: linker command failed with exit code 1 (use -v to see invocation)
[build] ninja: build stopped: subcommand failed.
[proc] The command: C:\msys64\clang64\bin\cmake.EXE --build E:/build-vita3k-clang-debug --parallel 15 --target all -- exited with code: 1
[driver] Build completed: 00:03:26.148
[build] Build finished with exit code 1
```
MSVC uses pragma like this `#pragma comment(lib, "Version.lib")`, but GCC and Clang do not understand this pragma.

Reply:
Indeed, then:
- Could you remove the pragma from the file where it can be done? if we are doing this in the cmake now
- Could you add the target_link_library to the relevant library using these functions instead of Vita3K? Fox example, the Version requirement is needed for the util library and DbgHelp to Tracy.

Reply:
1. Done.
2. Done. I have to add `Version.lib` to `vita3k` because `vita3k\util\src\vc_runtime_checker.cpp` belongs to `vita3k`.